projects/netgrif-components-core/src/lib/authentication/profile/services/profile.service.ts

```diff
@@ -0,0 +1,54 @@
+import { Injectable } from '@angular/core';
+import {ConfigurationService} from "../../../configuration/configuration.service";
+import {HttpClient} from "@angular/common/http";
+import {LoggerService} from "../../../logger/services/logger.service";
+import {Observable} from "rxjs";
+import {MessageResource} from "../../../resources/interface/message-resource";
+import {switchMap} from "rxjs/operators";
+import {processMessageResponse} from "../../../utility/process-message-response";
+import {UserChangePasswordRequest} from "../models/user-change-password-request";
+
+@Injectable({
+  providedIn: 'root'
+})
+export class ProfileService {
+
+    protected readonly _changePassword: string;
+
+
+    constructor(protected _config: ConfigurationService, protected _http: HttpClient, protected _log: LoggerService) {
+        const authAddress = _config.get().providers.auth.address;
```

Comment:
`.get()` makes a deep-copy of the configuration on each call. If only a subsection of the configuration is required there is a method for it in the service